Automattic/wp-calypso
# Store: Add delete to product category management

This PR hooks up deletion to the update screen. It adds some Redux code to make that possible. I also renamed `onTrash` for product categories to `onDelete`, because there is no concept of trash for categories in WooCommerce core, so it's unlikely that we would support trashing for categories any time soon.

Parents: #21351, #21379.
See: #20295

@kellychoffman @jameskoster Tagging both of you on this one and the saving one, since there is a little UI/interaction (even though they function like the other pages).

To Test:
* Boot up this branch and visit `http://calypso.localhost:3000/store/products/categories/:site`
* Select a category and then click "delete".
* You should see an accept notice. Accept it.
* Verify you see a success notice, are directed back to the listing, and that your category is removed from view.
* Run `npm run test-client client/extensions/woocommerce/state` and make sure all tests pass.
